syntax = "proto3";

package auth;

service AuthService {
  rpc Authenticate(AuthRequest) returns (AuthResponse);
  rpc RefreshToken(RefreshTokenRequest) returns (AuthResponse);
  rpc ExchangeToken(TokenExchangeRequest) returns (AuthResponse);
}

message AuthRequest {
  string username = 1;
  string password = 2;
}

message RefreshTokenRequest {
  string refresh_token = 1;
}

message TokenExchangeRequest {
  string token = 1;
}

message AuthResponse {
  string access_token = 1;
  string refresh_token = 2;
  int32 expires_in = 3;
  bool success = 4;
  string error_message = 5;
} 